09_employment/employment_metrics_county_all.qmd

```diff
@@ -0,0 +1,856 @@
+  scale_x_discrete(labels = label_wrap(10))
+
+```
+  
+* IPUMS documented values for cognitive_difficulty: 
+    + 0 N/A
+    + 1 No cognitive difficulty
+    + 2 Has cognitive difficulty 
+  
+Plot the options for cognitive_difficulty below.
+```{r}
+acs_clean %>% 
+  count(cognitive_difficulty) %>% 
+  ggplot(mapping = aes(x = cognitive_difficulty, y = n)) +
+  geom_col() +
+  theme_minimal() +
+  ggtitle("Cognitive difficulty, All Records") +
+ geom_text(mapping = aes(label = n), vjust = -1) +    
+  scale_y_continuous(expand = expansion(mult = c(0, 0.1))) +
+  labs(x = "cognitive_difficulty",
+       y = NULL)+
+  scale_x_discrete(labels = label_wrap(10))
+
+```
+
+Create the subgroup for disability. 
+```{r}
+acs_clean <- acs_clean %>% 
+  mutate(subgroup_disability = case_when(cognitive_difficulty == "Has cognitive difficulty" |
+                                           ambulatory_difficulty == "Has ambulatory difficulty" |
+                                           independent_living_difficulty == "Has independent living difficulty" |
+                                            vision_hearing_difficulty == "Has vision or hearing difficulty" |
+                                           self_care_difficulty == "Yes" ~ "with disabiliy",
+                                         TRUE ~ "without disability"))
+  
+```
+
+Check the variable breakout. 
+```{r}
+acs_clean %>% 
+  count(subgroup_disability) %>% 
+  ggplot(mapping = aes(x = subgroup_disability, y = n)) +
+  geom_col() +
+  theme_minimal() +
+  ggtitle("Subgroup Disability, All Records") +
+ geom_text(mapping = aes(label = n), vjust = -1) +    
+  scale_y_continuous(expand = expansion(mult = c(0, 0.1))) +
+  labs(x = "Disability",
+       y = NULL)+
+  scale_x_discrete(labels = label_wrap(10))
+```
+
+```{r}
+rm(acs_combined)
+
+```
+
+## Crosswalk
+
+Read in the PUMA to county crosswalk file. This file is created by the program generate_puma_county_crosswalks.rmd in the geographic-crosswalks folder. 
+
+```{r}
+puma_county_crosswalk <- read_csv(
+  here::here("geographic-crosswalks", "data", "crosswalk_puma_to_county.csv")
+)
+
+```
+
+Create a version of the crosswalk with just the county-level values of data quality variables for each crosswalk period.
+
+```{r}
+county_puma <- puma_county_crosswalk %>%
+  group_by(crosswalk_period, statefip, county) %>% 
+  summarise(
+    geographic_allocation_quality = mean(geographic_allocation_quality)
+  ) %>% 
+  ungroup()
+
+```
+
+## Merge Crosswalk 
+
+Create a variable in the ACS data indicating whether the survey is pre or post-2022. This will be used to join on the correct crosswalk information.
+
+```{r}
+acs_clean <- acs_clean %>% 
+  mutate(crosswalk_period = ifelse(year < 2022, "pre-2022", "2022"))
+
+```
+
+Check that the assignment worked. All years prior to 2022 should be designated "pre-2022".
+
+```{r}
+count(acs_clean, crosswalk_period, year)
+
+```
+
+Join the cleaned ACS data onto the crosswalk.
+
+```{r}
```

Comment:
For this case I'm going to add a join onto the crosswalk file later on in the program so that we guarantee the count of counties lines up. This join is so that we have the "county" varaible filled in for every PUMA in the micro data.